docs/ac-remote-update/how-it-works.md

```diff
@@ -0,0 +1,61 @@
+
+We could receive a remote configuration with an older Agent Control version which doesn't support this feature. In that situation, the Agent Control would end up in a "non-recoverable" state. Updating it would require access to the system. Ideally, the backend should prevent that.
+
+##### Updating Flux version
+
+Flux it's a project of the [CNCF](https://www.cncf.io/) with their own roadmap. Thus, we have to be aware of breaking changes introduced in [flux2](https://github.com/fluxcd/flux2), and potential compatibility issues with the Kubernetes version that we are using. That information is usually in the changelog. For example, [v2.6.0](https://github.com/fluxcd/flux2/releases/tag/v2.6.0). **Migrations** are not automated.
+
+We could receive a remote config that installs Flux and breaks Agent Control because some breaking changed where introduced to the [Custom Resource Definition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions).
+
+## How it works
+
+The high level overview of the happy path is simple:
+
+1. Fleet Control sends a new remote config with the updated version through OpAMP
+2. Agent Control receives and validates the configuration
+3. Agent Control applies the update on the system
+
+The implementation is, of course, more complex.
+
+### Kubernetes
+
+Let's start with the Agent Control update.
+
+![](./images/update-ac.png)
+
+1. **Receive Configuration**: The process begins when a new remote config is received in the `process_events` function, which then passes it to `handle_remote_config` for processing.
+2. **Report Initial State**: To provide visibility, Agent Control reports its status as `Applying`.
+3. **Validate and Apply**: The new configuration is validated and stored by the `validate_apply_store_remote_config` function. On success, this function updates Agent Control's own `HelmRelease` with the new version.
+4. **Flux**
+4. **Report Final State**: Once the configuration has been successfully applied to the HelmRelease object, Agent Control reports its status as `Applied`.
+5. **Reconcile in Cluster**: Flux’s Helm Controller detects the change to the `HelmRelease` and begins a rolling update of the Agent Control pods to finalize the process.
+
+If there's any error, Agent Control will send `Failed` status.
+
+![](./images/update-ac-sequence.png)
+
+The rolling update is an [strategy](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy) implemented by kubernetes that incrementally replaces old pods with new ones. It's done in such a way, that old pods will be kept alive if new ones can't be started. Kubernetes assures it with [Liveness and Startup probes](https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/).
+
+![](./images/rolling-update.png)
+
+Flux update works the same way. The only difference is that the `HelmRelease` will be reconciled into itself.
+
+![](./images/update-flux.png)
```

Comment:
As both work the same way, would it make sense to have only one diagram?

Reply:
I thought about that, because it's only one extra line. At the same time, I don't know if this makes it easier for a new person to see which is the actual difference in both flows.

Reply:
Done